### PR TITLE
core: Truncate socket host by null bytes

### PR DESCRIPTION
**Pull request by Dinnerbone**
Example:
```actionscript
socket.connect("127.0.0.1" + unescape("%00") + "this part is ignored, flash stops at the null byte")
```

Flash truncates at any null byte - likely a low level method thinking it's getting a cstring. This emulates the truncation behaviour, fixing a Habbo connection issue from Discord.

No tests because our test suite won't run into this, it ignores hosts and everything is local.

### CODE REVIEW

**Review by adrian17 — APPROVED** [review]
Can you add even just a small local unit `#[test]` with the `"127.0.0.1\0nonsense` example?

**Comment by Dinnerbone** [comment]
Added unit test!

File: core/src/socket.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ use crate::{
 use async_channel::{unbounded, Receiver, Sender as AsyncSender, Sender};
 use gc_arena::Collect;
 use slotmap::{new_key_type, SlotMap};
+use std::ops::Deref;
```
Comment (danielhjacobs) [inline core/src/socket.rs:17]: CI seems to think this is unnecessary.

File: core/src/socket.rs [hunk 2]
```diff
@@ -435,3 +436,29 @@ impl<'gc> Sockets<'gc> {
         }
     }
 }
+
+/// Flash treats a socket host as a cstring, and stops reading at a null byte.
+/// We need to account for this here.
+fn sanitize_host(host: &str) -> &str {
+    host.split('\0').next().unwrap_or(host)
```
Comment (evilpie) [inline core/src/socket.rs:443]: AFAIK Just `unwrap` should be fine.

**Review by sleepycatcoding — APPROVED** [review]

**Pull request merged**